cli2gui/application/application.py

```diff
@@ -3,22 +3,89 @@
 # pylint: disable=import-outside-toplevel
 from __future__ import annotations
 
+# TODO? https://github.com/syldium/pyterminal
+
+# TODO unused parts of the window should be grabbable (click + move = move window)
+
+# TODO switch tabs by scrolling (this should be a feature of pysimplegui)
+
+# TODO make the window easier to kill. currently i need *two* kill signals
+
+# TODO window should always fit on screen. currently the window grows larger than screen
+# when there are too many options. should have vertical-scroll, like Gooey.
+# scrollale columns: https://github.com/PySimpleGUI/PySimpleGUI/issues/1779
+## unbind mouse-wheel from combobox
+#window['Combobox'].TKCombo.unbind_class("TCombobox", "<MouseWheel>")
+
+# TODO? responsive layout https://github.com/PySimpleGUI/PySimpleGUI/issues/4597
+# https://github.com/PySimpleGUI/PySimpleGUI/issues/3921
+
+# TODO fix title + subtitle
+# should expand to full width
+
+# FIXME "nano" editor is broken -> bug in pyte?
+# bug: enter does not produce a new line
+# maybe we must send \r\n not \n
+
+# TODO terminal colors
+# color test programs:
+#   msgcat --color=test
+#   colors=256; for (( n=0;n<colors;n++ )) do printf "$(tput setaf $n)%3s$(tput sgr0) " $n; (( (n+3)%6==0 )) && echo; done
+#   colors=256; for (( n=0;n<colors;n++ )) do printf "$(tput setaf $n)%3s$(tput sgr0) " $n; (( (n+1)%16==0 )) && echo; done
+
+# FIXME "nyancat" program is not working
+# i only see:
+# You have nyaned for 2 seconds!
+# You have nyaned for 14 seconds!
+# You have nyaned for 22 seconds!
+# ...
+
+# TODO use camelCase for all vars
+
 import json
 import sys
 from pathlib import Path
 from typing import Any
+import subprocess
+import threading
+import os
+import select
+import pty
+import shlex
+import logging
+import io
+import time
+import re
+import errno
+
+import psutil
+
+# short loglevel names
+# https://github.com/python/cpython/blob/677320348728ce058fa3579017e985af74a236d4/Lib/logging/__init__.py#L100
+#logging._levelToName = { 0: 'N', 10: 'D', 20: 'I', 30: 'W', 40: 'E', 50: 'C' }
+#logging_levels_pretty = ["Debug", "Info", "Warning", "Error", "Critical"]
+
+import tkinter as tk
+import pyte # terminal emulator. render terminal output to visible characters
 
 try:
 	from getostheme import isDarkMode
 except ImportError:
 	isDarkMode = lambda: True
 import yaml
-from PySimpleGUI import Element, Window
+
+#from PySimpleGUI import Element, Window
+from .PySimpleGUI import Element, Window # local version
 
 from .. import c2gtypes
 from .pysimplegui2args import argFormat
 from .widgets import Widgets
 
+# TODO(milahu) better ...
+terminal_width = 80
+terminal_height = 24
+num_history_lines = 100 # aka "scrollback buffer size"
```

Comment:
Can you try and stick by the current style guide please? Not a problem at all for a draft but this needs fixing before a merge :)